programs/governance-rewards/src/instructions/create_distribution.rs

```diff
@@ -0,0 +1,89 @@
+use anchor_lang::prelude::*;
+use std::mem::size_of;
+
+use crate::{
+    error::GovernanceRewardsError,
+    state::{distribution::Distribution, distribution_option::DistributionOptions},
+};
+
+/**
+ * Instruction to create a Distribution.
+ *
+ * The caller must provide a registration_cutoff, which specifies the timestamp at
+ * which registration will end.
+ */
+#[derive(Accounts)]
+pub struct CreateDistribution<'info> {
+    /**
+     * Address of the distribution to be created.
+     */
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + size_of::<Distribution>()
+    )]
+    pub distribution: Box<Account<'info, Distribution>>,
+
+    /**
+     * Account to own any provided token accounts.
+     */
+    /// CHECK: Not read
+    #[account(seeds = [b"payout authority".as_ref(), distribution.key().as_ref()], bump)]
+    pub payout_authority: AccountInfo<'info>,
+
+    /**
+     * Realm to which the distribution belongs.
+     */
+    /// CHECK: Not read
+    pub realm: AccountInfo<'info>,
+
+    /**
+     * Choice of program to create voter weight records.
+     *
+     * The program will check whether presented voter weight records are owned by this
+     * account. For example, you might provide the address of the voter stake registry
+     * program.
+     */
+    /// CHECK: Not read
+    pub voter_weight_program: AccountInfo<'info>,
+
+    /// CHECK: Not read
+    #[account(mut)]
+    pub payer: AccountInfo<'info>,
```

Comment:
doesn't the payer need to be a signer, too?

Reply:
I dunno, it seems to be working without the payer as a signer

Reply:
I guess that's accidental? Maybe in your tests the payer is also the fee-payer, or the admin? To get the constraints right and let anchor do the right thing payers should be signers.